src/charm.py

```diff
@@ -137,8 +137,19 @@ def _configure(self, _):
             container.restart(self._name)
             logger.info("Prometheus (re)started")
 
+        # Provide connection info to any remote write clients.
+        self._set_remote_write()
+
         self.unit.status = ActiveStatus()
 
+    def _set_remote_write(self):
+        """Provide connection info to any remote write clients."""
+        if self.model.relations["ingress"] and self.app.planned_units() == 1:
```

Comment:
It is not obvious why you need `self.planned_units()` ? Could you kindly elaborate upon this in docstring or code comment.

Reply:
Oh yes I will add a comment. Ingress won't work with more than one unit. I will go in to more depth in the comment.

Reply:
If it will not work with **more** than one unit shouldn't you be checking the total number of units rather than just "planned_unit" . Also this is quite surprising. I do not see why Ingress will not work with more than one unit. Pray, illuminate me and help me get rid of my ignorance here.

Reply:
> Ingress won't work with more than one unit

In that case I would add a leader guard - for a less messy scaledown.
But this is a suspicious limitation. What is supposed to happen to prometheus unit count in HA?

Reply:
Ah I see. You mean with more than one unit of Prometheus. Indeed. In that case how are you informing the system administrator that something has gone wrong and he/she needs to do something about it ?

Reply:
The issue here is that ingress doesn't allow you to address individual pods which remote write requires. I don't see how a leader guard would have anything to do with this. The issue with detecting the issue is there is no way to determine if the related app can route to you. I will put a warning in the logs but I'm afraid that is all that can be done without resorting to messy config variables.

Reply:
I would set BlockedStatus since the admin is more likely to view juju status rather than logs. But more importantly there seems the need to also put a TODO code comment here and create a github issue. This is because as I understand you are saying this is an incomplete solution that only works with a single mongodb unit and breaks down the moment you scale up. This may not be a solveable problem now and may require support from Juju/Ops but at least we should note it down that we have this technical debt lest we forget.
@mmanciop  WDYT.

Reply:
Might be worth it to look into improving the ingress charm.